### DISCUSSION TITLE
In NextJS api route, When `return NextResponse.json()` in callback function that return does not working.

### DISCUSSION THREAD

**Discussion by clevermiraz** [vercel/next.js#58479] [App Router]
```js
axios.post(apiUrl, postData)
  .then(response => {
    // Handle successful response here
    console.log('POST request successful:', response.data);
    return NextResponse.json(data: response.data)
  })
  .catch(error => {
    // Handle error here
    console.error('Error making POST request:', error);
    return NextResponse.json(message: "error")
  });
  ```
  
  this is example code in that code the return function not working in callback function.
  why this happend?

**Comment by lixy3** [comment]
I am encountering the same issue. Could you please let me know if you managed to fix it meanwhile?

**Comment by marc-ed-raffalli** [comment]
Would you share a bit more details aout the request handler? 
I would first suspect the promise returned by axios (resolving the `NextResponse`) is missing a await or return
